src/coreclr/src/pal/inc/pal.h

```diff
@@ -1108,11 +1099,7 @@ OpenSemaphoreW(
     IN LPCWSTR lpName);
 
 #ifdef UNICODE
-#define CreateSemaphore CreateSemaphoreW
-#define CreateSemaphoreEx CreateSemaphoreExW
-#else
-#define CreateSemaphore CreateSemaphoreA
-#define CreateSemaphoreEx CreateSemaphoreExA
+#define CreateSemaphore CreateSemaphoreExW
```

Comment:
This is still missing `Ex`

Reply:
On callsites it is either `CreateSemaphoreExW` or `CreateSemaphore`. There is no usage left of `CreateSemaphoreEx`.

Reply:
Basically we need to delete `#ifdef UNICODE` condition, as there is only one API for each.

Reply:
Where does it not work?

The PAL is meant to emulate Windows. Windows has this under `#ifdef UNICODE`.

Reply:
When we removed CreateA CreateExA functions, the #else // UNICODE case was blanked out. Without this condition everything goes green. With the condition, it fails for the paths where UNICODE is not set.

> Windows has this under `#ifdef UNICODE`

Does Windows also has the `#else` case? If maintaining the compat with windows is priority then I think we should keep CreateA functions too. Otherwise, we can treat all code paths leading to these as wide on unix.

Reply:
Ah ok - make sense. Thank you for being patient with me :-)